nix 0.22 deprecated as_errno and made it into a no-op
nix 0.26 removed it.

nix 0.27 stopped enabling stuff by default, requring features to be enabled
explicitly. These features existed starting in nix 0.24.

Index: libmount/Cargo.toml
===================================================================
--- libmount.orig/Cargo.toml
+++ libmount/Cargo.toml
@@ -28,7 +28,8 @@ path = "src/lib.rs"
 version = "0.2.28"
 
 [dependencies.nix]
-version = "0.14"
+version = ">= 0.24, < 1.0"
+features = ["user", "mount"]
 
 [dependencies.quick-error]
 version = "2.0"
Index: libmount/src/lib.rs
===================================================================
--- libmount.orig/src/lib.rs
+++ libmount/src/lib.rs
@@ -85,7 +85,7 @@ impl OSError {
     fn from_nix(err: nix::Error, explain: Box<Explainable>) -> OSError {
         OSError(
             MountError::Io(
-                err.as_errno().map_or_else(|| io::Error::new(io::ErrorKind::Other, err), io::Error::from),
+                io::Error::from(err),
             ),
             explain,
         )
